libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1010,14 +1010,15 @@ export class RushConfiguration {
   public static tryFindRushJsonLocation(options?: ITryFindRushJsonLocationOptions): string | undefined {
     const optionsIn: ITryFindRushJsonLocationOptions = options || {};
     const verbose: boolean = optionsIn.showVerbose || false;
-    let currentFolder: string = optionsIn.startingFolder || process.cwd();
+    const currentFolder: string = optionsIn.startingFolder || process.cwd();
 
-    // Look upwards at parent folders until we find a folder containing rush.json
-    for (let i: number = 0; i < 10; ++i) {
+    // Recursively look upwards at parent folders until we find a folder containing rush.json,
+    // or we reach the root directory without finding a rush.json file
+    function _tryFindRushJsonLocation(_currentPage: string) {
```

Comment:
This would be more efficient if it was written iteratively.

Reply:
I thought that using recursion would make the logic clearer, but I think you are right. I have switched to using a while loop instead.